x-pack/plugins/maps/public/actions/data_request_actions.ts

```diff
@@ -162,19 +170,24 @@ function syncDataForAllJoinLayers() {
         return 'hasJoins' in layer ? (layer as IVectorLayer).hasJoins() : false;
       })
       .map((layer) => {
-        return dispatch(syncDataForLayer(layer));
+        return dispatch(syncDataForLayer(layer, false, isForceRefresh));
       });
     await Promise.all(syncPromises);
   };
 }
 
-export function syncDataForLayer(layer: ILayer, forceRefresh: boolean = false) {
+export function syncDataForLayer(
```

Comment:
Would it make the method definition clearer to put all arguments in an object? Or at least forceRefreshDueToDrawing and isForceRefresh. I think this could be an easy place for errors since the method definition takes two boolean parameters and it is easy to reverse them.

Maybe something like 
```
export function syncDataForLayer({
  layer,
  forceRefreshDueToDrawing,
  isForceRefresh
}: {
  layer: ILayer;
  forceRefreshDueToDrawing: boolean;
  isForceRefresh: boolean;
})
```

or 

```
export function syncDataForLayer({
  layer,
  forceRefresh,
}: {
  layer: ILayer;
  forceRefresh: {
    forceRefreshDueToDrawing: boolean;
    isForceRefresh: boolean;
  }
})
```

Reply:
Actually another thought, instead of polluting `syncDataForLayer` with a complicated method signature. How about just making methods: `syncDataForLayer` and `syncDataForLayerDueToDrawing`?

Reply:
+1 on the addition of the new `syncDataForLayerDueToDrawing` function.